File: plugins/dotnet/robocode.dotnet.control.api/src/implementation/robocode/control/snapshot/IScoreSnapshot.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
﻿

using System;
using System.Collections.Generic;
using System.Text;

namespace Robocode.Control.Snapshot
{
    /// <summary>
    /// Interface of a bullet snapshot at a specific time in a battle.
    /// </summary>
    public interface IScoreSnapshot : IComparable<IScoreSnapshot>
    {
        /// <summary>
        /// Contains the name of the contestant, i.e. a robot or team.
        /// </summary>
        /// <value>
        /// The name of the contestant, i.e. a robot or team.
        /// </value>
        string Name { get; }

        /// <summary>
        /// Contains the total score.
        /// </summary>
        /// <value>
        /// The total score.
        /// </value>
        double TotalScore { get; }

        /// <summary>
        /// Contains the total survival score.
        /// </summary>
        /// <value>
        /// The total survival score.
        /// </value>
        double TotalSurvivalScore { get; }

        /// <summary>
        /// Contains the total last survival score.
        /// </summary>
        /// <value>
        /// The total last survival score.
        /// </value>
        double TotalLastSurvivorBonus { get; }

        /// <summary>
        /// Contains the total bullet damage score.
        /// </summary>
        /// <value>
        /// The total bullet damage score.
        /// </value>
        double TotalBulletDamageScore { get; }

        /// <summary>
        /// Contains the total bullet kill bonus.
        /// </summary>
        /// <value>
        /// The total bullet kill bonus.
        /// </value>
        double TotalBulletKillBonus { get; }

        /// <summary>
        /// Contains the total ramming damage score.
        /// </summary>
        /// <value>
        /// The total ramming damage score.
        /// </value>
        double TotalRammingDamageScore { get; }

        /// <summary>
        /// Contains the total ramming kill bonus.
        /// </summary>
        /// <value>
        /// The total ramming kill bonus.
        /// </value>
        double TotalRammingKillBonus { get; }

        /// <summary>
        /// Contains the total number of first places.
        /// </summary>
        /// <value>
        /// The total number of first places.
        /// </value>
        int TotalFirsts { get; }

        /// <summary>
        /// Contains the total number of second places.
        /// </summary>
        /// <value>
        /// The total number of second places.
        /// </value>
        int TotalSeconds { get; }

        /// <summary>
        /// Contains the total number of third places.
        /// </summary>
        /// <value>
        /// The total number of third places.
        /// </value>
        int TotalThirds { get; }

        /// <summary>
        /// Contains the current score.
        /// </summary>
        /// <value>
        /// The current score.
        /// </value>
        double CurrentScore { get; }

        /// <summary>
        /// Contains the current survival score.
        /// </summary>
        /// <value>
        /// The current survival score.
        /// </value>
        double CurrentSurvivalScore { get; }

        /// <summary>
        /// Contains the current survival bonus.
        /// </summary>
        /// <value>
        /// The current survival bonus.
        /// </value>
        double CurrentSurvivalBonus { get; }

        /// <summary>
        /// Contains the current bullet damage score.
        /// </summary>
        /// <value>
        /// The current bullet damage score.
        /// </value>
        double CurrentBulletDamageScore { get; }

        /// <summary>
        /// Contains the current bullet kill bonus.
        /// </summary>
        /// <value>
        /// The current bullet kill bonus.
        /// </value>
        double CurrentBulletKillBonus { get; }

        /// <summary>
        /// Contains the current ramming damage score.
        /// </summary>
        /// <value>
        /// The current ramming damage score.
        /// </value>
        double CurrentRammingDamageScore { get; }

        /// <summary>
        /// Contains the current ramming kill bonus.
        /// </summary>
        /// <value>
        /// The current ramming kill bonus.
        /// </value>
        double CurrentRammingKillBonus { get; }
    }
}

File: robocode.api/src/main/java/robocode/control/snapshot/IScoreSnapshot.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package robocode.control.snapshot;


/**
 * Interface of a score snapshot at a specific time in a battle.
 * 
 * @author Pavel Savara (original)
 * @author Flemming N. Larsen (contributor)
 *
 * @since 1.6.2
 */
public interface IScoreSnapshot extends Comparable<Object> {

	/**
	 * Returns the name of the contestant, i.e. a robot or team.
	 *
	 * @return the name of the contestant, i.e. a robot or team.
	 */
	String getName();

	/**
	 * Returns the total score.
	 *
	 * @return the total score.
	 */
	double getTotalScore();

	/**
	 * Returns the total survival score.
	 *
	 * @return the total survival score.
	 */
	double getTotalSurvivalScore();

	/**
	 * Returns the total last survivor score.
	 *
	 * @return the total last survivor score.
	 */
	double getTotalLastSurvivorBonus();

	/**
	 * Returns the total bullet damage score.
	 *
	 * @return the total bullet damage score.
	 */
	double getTotalBulletDamageScore();

	/**
	 * Returns the total bullet kill bonus.
	 *
	 * @return the total bullet kill bonus.
	 */
	double getTotalBulletKillBonus();

	/**
	 * Returns the total ramming damage score.
	 *
	 * @return the total ramming damage score.
	 */
	double getTotalRammingDamageScore();

	/**
	 * Returns the total ramming kill bonus.
	 *
	 * @return the total ramming kill bonus.
	 */
	double getTotalRammingKillBonus();

	/**
	 * Returns the total number of first places.
	 *
	 * @return the total number of first places.
	 */
	int getTotalFirsts();

	/**
	 * Returns the total number of second places.
	 *
	 * @return the total number of second places.
	 */
	int getTotalSeconds();

	/**
	 * Returns the total number of third places.
	 *
	 * @return the total number of third places.
	 */
	int getTotalThirds();

	/**
	 * Returns the current score.
	 *
	 * @return the current score.
	 */
	double getCurrentScore();

	/**
	 * Returns the current survival score.
	 *
	 * @return the current survival score.
	 */
	double getCurrentSurvivalScore();

	/**
	 * Returns the current survival bonus.
	 *
	 * @return the current survival bonus.
	 */
	double getCurrentSurvivalBonus();

	/**
	 * Returns the current bullet damage score.
	 *
	 * @return the current bullet damage score.
	 */
	double getCurrentBulletDamageScore();

	/**
	 * Returns the current bullet kill bonus.
	 *
	 * @return the current bullet kill bonus.
	 */
	double getCurrentBulletKillBonus();

	/**
	 * Returns the current ramming damage score.
	 *
	 * @return the current ramming damage score.
	 */
	double getCurrentRammingDamageScore();

	/**
	 * Returns the current ramming kill bonus.
	 *
	 * @return the current ramming kill bonus.
	 */
	double getCurrentRammingKillBonus();
}
